From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Tue, 1 Sep 2020 20:30:45 -0700
Subject: fix: passing off_t* to ssize_t* param

Fixes incompatible pointer types error where uv__fs_sendfile was trying
to pass 'off_t *' (aka 'long long *') to parameter of type 'ssize_t *'
(aka 'int *') [-Werror,-Wincompatible-pointer-types]
in its invoction of:

r = uv__fs_copy_file_range(in_fd, NULL, out_fd, &off, req->bufsml[0].len, 0);

diff --git a/deps/uv/src/unix/fs.c b/deps/uv/src/unix/fs.c
index 87cb8b816aea3942f8a15cf76e8da909d79e556f..4e1be035387dc9225eafebff11018fd454c2c7c0 100644
--- a/deps/uv/src/unix/fs.c
+++ b/deps/uv/src/unix/fs.c
@@ -893,7 +893,7 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
       static int copy_file_range_support = 1;
 
       if (copy_file_range_support) {
-        r = uv__fs_copy_file_range(in_fd, NULL, out_fd, &off, req->bufsml[0].len, 0);
+        r = uv__fs_copy_file_range(in_fd, NULL, out_fd, &((ssize_t)off), req->bufsml[0].len, 0);
 
         if (r == -1 && errno == ENOSYS) {
           errno = 0;
